stats/stats.go

```diff
@@ -261,6 +261,25 @@ func (st *SampleTags) CloneTags() map[string]string {
 	return res
 }
 
+// ReplaceURLWithName replaces url tag with name tag, if both exist.
+// This method should on call for cloud output usage, see Issue #1166.
+func (st *SampleTags) ReplaceURLWithName() {
+	if st == nil || st.tags == nil {
+		return
+	}
+
+	name, nameExist := st.tags["name"]
+	_, urlExist := st.tags["url"]
+	if nameExist && urlExist {
+		st.tags["url"] = name
+		res, err := json.Marshal(st.tags)
```

Comment:
No need to prematurely call `json.Marshal()` here. See how `SampleTags` has a `MarshalJSON()` method: https://github.com/loadimpact/k6/blob/c2d73ba070b5223d60e784401e9bda8d634440fe/stats/stats.go#L229-L235

It it just uses the json property to cache the result. But with your code, if the `json` properly is already defined, it will just be returned here and you'd save it again with `st.json = res` :smile: So, instead of all of this, here you should just set `st.json = nil`

Reply:
Fixed.